Instrukcije_d001/src/app/terminal.c

```diff
@@ -4,54 +4,86 @@
+        .maxY = y,
+        .steviloPolja = 0
+    };
 
-    noecho();
-    refresh();
+    return this;
+}
+
+void terminal_validate_input(World* world, Terminal_input* input) {
+    if (input->input >= 0 && input->input <= 8) { // input je veljaven
+        if (world->memoTabela[input->x][input->y] != 1 && world->memoTabela[input->x][input->y] != 2) { // še mismo dodal sem
+            world_poteza(world, world->memoTabela, input->input); // torej zgodila se je poteza
+        } else { // input, ki je ze bil izbran
+            mvaddstr(10, 0, "Your input has already been chosen, please try again!");
+        }
+    }
+
+    if (input->input < 0 || input->input > 8)
+        mvaddstr(10, 0, "Your input is invalid, please try again!");
+}
+
+Terminal_input terminal_get_input(World* world) {
+    mvaddstr(0, 0, "Enter a number between 0 and 8 to add your piece:");
+    int vnos = getch() - '0'; // read the players input
+
+    Terminal_input input = terminal_input_new(vnos, vnos / WORLD_SIRINA_BOARDA, vnos % WORLD_SIRINA_BOARDA);
+
+    mvaddch(0, 50, vnos + '0'); // za prikazat userjev input
+
+    // to je za popucat napačne inpute (šele naslednjo iteracijo)
+    move(10, 0); clrtoeol();
+
+    terminal_validate_input(world, &input); // ta funkcija preveri, da je vse v redu in če ni, to izpiše
+
+    return input;
+}
+
+Terminal_input terminal_input_new(int input, int x, int y) {
+    Terminal_input this;
+
+    this = (Terminal_input){
+        .input = input,
+        .x = x,
+        .y = y
+    };
 
     return this;
 }
 
 int terminal_main() {
     World world = world_new();
-    Terminal terminal = terminal_init();
 
     while (1) {
         terminal_draw_world(&world);
 
-        mvaddstr(0, 0, "Enter a number between 0 and 8 to add your piece:");
-        terminal.input = getch() - '0'; // read the players input
-
-        terminal_izpisi_in_pucanje(terminal.input);
-
-        int x = terminal.input / SIRINA_BOARDA; // nadstropje memoTabela
-        int y = terminal.input % SIRINA_BOARDA; // soba v nadstropju v memoTabela
-
-        if (terminal.input >= 0 && terminal.input <= 8) { // input je veljaven
-            if (world.memoTabela[x][y] != 1 && world.memoTabela[x][y] != 2) { // še mismo dodal sem
-                poteza(&world, world.memoTabela, terminal.input); // torej zgodila se je poteza
-            } else { // input, ki je ze bil izbran
-                mvaddstr(10, 0, "Your input has already been chosen, please try again!");
-            }
-        }
+        Terminal_input input = terminal_get_input(&world);
 
         if (world.active == 1)
             break;
 
         // (pri tem if-u moramo dati == 1, ker če ne bo tudi v primeru drawa razlicno od nic in bo slo v if in ne else if)
-        if (dinamicnoPoisciZmagovalca(x, y, (world.frames % 2 == 0) ? 1 : 2, world.memoTabela, &world) == 1) {
+        if (world_dinamicnoPoisciZmagovalca(&world, input.x, input.y, (world.frames % 2 == 0) ? 1 : 2, world.memoTabela) == 1) {
```

Comment:
world_dinamicnoPoisciZmagovalca kliči samo enkrat in rezultat shrani v spremenljivko ter potem to spremenljivko uporabljaj v svoji kodi. Pa preden se to lotiš v world strukturi se ti nahaja kdo je na vrsti tako da  `(world.frames % 2 == 0) ? 1 : 2` prestavi kar v to funkcijo `world_dinamicnoPoisciZmagovalca ` glej komentar v `world.h`